src/main/java/uk/gov/digital/ho/hocs/casework/api/DaysElapsedCalculatorImpl.java

```diff
@@ -1,42 +1,34 @@
 package uk.gov.digital.ho.hocs.casework.api;
 
-import com.fasterxml.jackson.databind.ObjectMapper;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
 import org.springframework.util.StringUtils;
-import uk.gov.digital.ho.hocs.casework.domain.model.StageWithCaseData;
 
 import java.time.LocalDate;
 import java.time.format.DateTimeFormatter;
-import java.util.HashMap;
 import java.util.Map;
 
 @Service
 public class DaysElapsedCalculatorImpl implements DaysElapsedCalculator {
 
-    private WorkingDaysElapsedProvider workingDaysElapsedProvider;
-    private ObjectMapper objectMapper;
+    private final WorkingDaysElapsedProvider workingDaysElapsedProvider;
 
     private static final String DATE_RECEIVED_FIELD_NAME = "DateReceived";
     private static final String DATE_FORMAT = "yyyy-MM-dd";
 
     @Autowired
-    public DaysElapsedCalculatorImpl(WorkingDaysElapsedProvider workingDaysElapsedProvider, ObjectMapper objectMapper) {
+    public DaysElapsedCalculatorImpl(WorkingDaysElapsedProvider workingDaysElapsedProvider) {
         this.workingDaysElapsedProvider = workingDaysElapsedProvider;
-        this.objectMapper = objectMapper;
     }
 
     @Override
-    public void updateDaysElapsed(StageWithCaseData stage) {
-        Map<String, String> data = new HashMap<>(stage.getDataMap(objectMapper));
+    public void updateDaysElapsed(Map<String, String> data, String caseType) {
         int daysElapsed = 0;
         String dateString = data.get(DATE_RECEIVED_FIELD_NAME);
         if (StringUtils.hasText(dateString)) {
             LocalDate dateToCheck = LocalDate.parse(dateString, DateTimeFormatter.ofPattern(DATE_FORMAT));
```

Comment:
The amount of places we parse dates, I wonder if having this in it's own component would help...

Reply:
👀  funny you should say that